frc_characterization/logger_gui/__init__.py

```diff
@@ -132,12 +136,15 @@ def disableTestButtons():
 
     def enableTestButtons():
         all_completed = True
-        for step in tests[:-1]:
-            step.enable()
-            all_completed &= step.isCompleted()
-        # Don't have to do trackwidth if not drivetrain
-        if STATE.test == "Drivetrain":
-            all_completed &= tests[-1].isCompleted()
+        for step in tests:
+            # Don't have to do trackwidth if not drivetrain
+            if step.getName() != "Trackwidth":
+                step.enable()
+                all_completed &= step.isCompleted()
+            else:
+                if STATE.test == "Drivetrain":
+                    step.enable()
+                    all_completed &= tests[-1].isCompleted()
```

Comment:
This still has the same problem as before, what is `tests[-1]` -> not always guaranteed to be the drivetrain test?

I don't think atleast?

Reply:
Yeah I forgot to change that.

On a similar note, I also found this method that was relying on `tests[-1]`:
https://github.com/Piphi5/frc-characterization/blob/Refactor/frc_characterization/logger_gui/__init__.py#L217-#L225

Do you know a way of finding the trackwidth test that doesn't require looping through the list of tests?

Reply:
Nope, unless you store them in a hashmap somewhere.

TBH, we're not going to have 3000 different tests so the "performance" here probably won't matter much.

Reply:
[Initial Issue fixed](https://github.com/Piphi5/frc-characterization/blob/Refactor/frc_characterization/logger_gui/__init__.py#L146)

And [additional issue also fixed](https://github.com/Piphi5/frc-characterization/blob/Refactor/frc_characterization/logger_gui/__init__.py#L217-#L231)

Reply:
This feels weird to me still. I guess it works, but why do we have two functions which seem to perform nearly the same thing: https://github.com/Piphi5/frc-characterization/blob/08295a17a3b0c5b64b7c7554f7126b1f1b9c8ce2/frc_characterization/logger_gui/__init__.py#L126-L127
and they're both called one after another?

The `changeTestButtons` function also does it "nicer"; one for loop over the elements and enabling when it hits it.

Reply:
It seems like the enableTestButtons serves to enable the buttons (after a logger run is over, etc.) and check to see if all the necessary tests had been performed. The changeTestButtons takes care of the case where the user switches the logger test to a Drivetrain Test and needs trackwidth to be enabled. 

So functionally they accomplish two different tasks, but it may be possible to get rid of the changeTestButton. Would that be a good idea?

Reply:
They seem like mostly the same code... so maybe it could be abstracted a little?

Reply:
I'm having a hard time identifying what possible snippets could be abstracted. Do you have any suggestions?

Reply:
Yeah, not sure.